### ISSUE TITLE
init_enum cause server crash?

### ISSUE THREAD

**Issue by havena** [CBATeam/CBA_A3#126]
since a3 1.48 / ace 3.1.1 / cba rc7 i have server crash every day.
dwarden send me a verbose cba_xeh with debug diag_log and this is the RPT results on Crash :

```
22:41:18 init_enum: _cfgEntry = bin\config.bin/Extended_InitPost_EventHandlers/All/ace_common_executePersistent
=======================================================
-------------------------------------------------------
Exception code: C0000005 ACCESS_VIOLATION at 00B5BD0C
```
```
23:20:41 init_enum: _cfgEntry = bin\config.bin/Extended_InitPost_EventHandlers/All/ace_common_executePersistent
=======================================================
-------------------------------------------------------
Exception code: C0000005 ACCESS_VIOLATION at 00B5BD0C
```

some rpt https://mega.co.nz/#!ftIj3YhT!1me7_USgfQemao8iOL1CB_kbkV3e0stG6lW47ovfLRw

**Comment by havena** [comment]
i have add a diag_log debug in ace_common_executePersistent, but nothing was call by this fnc... 

```
// by commy2
#include "script_component.hpp"

PARAMS_1(_target);

{
    if (isNil "_x") then {
        diag_log text format ["[ACE] ERROR: No argument and function for remote function. ID: %1", _forEachIndex];
    } else {
        if (typeName _x == "ARRAY") then {
			diag_log format ["DEBUG ACE executePersistent === %1 - %2 call %3",_x select 0,_target,_x select 1];
            [_x select 0, _target] call (_x select 1);
        };
    };
} forEach (_target getVariable ["ACE_PersistentFunctions", []]);
```

**Comment by Killswitch00** [comment]
This is a very difficult thing to analyze. From what Dwarden has told us, the crash happens when the init_enum.sqf script calls the `configName _cfgEntry` command, but the value of _cfgEntry just before the crash is perfectly valid (and looking at the log, it has been called before with that same value)

* Do the crashes happen every time you run the server?
* At what point does the server crash?
* Is there a difference in crashing depending on how many players are connected? 
* If you remove `cba_enable_auto_xeh.pbo`, does the server still crash?

**Comment by havena** [comment]
server crash when player > ~80.

i havent try without cba_enable_auto_xeh.pbo cause i have a bug with interaction without. i dont find how to fix it.

**Comment by robalo** [comment]
Noticed in RPT -maxmem=2047 was set. Are you monitoring the memory usage ? Could be worth to try upping or removing that limit if you're getting close to reaching it.

**Comment by havena** [comment]
no, 2000 and 1800 on last crashs, but lot Memory PageFault before crash

**Comment by havena** [comment]
since 1.48 fps/cps are low, crash happend when cps/fps like ~7/9 @80 players.
with older arma version server can handle 20fps @120 players.

**Comment by havena** [comment]
robalo commented 2 hours ago
Noticed in RPT -maxmem=2047 was set. Are you monitoring the memory usage ? Could be worth to try upping or removing that limit if you're getting close to reaching it.
```
256 is hard-coded minimum (anything lower falls backs to 256). 2047 is hard-coded maximum (anything higher falls back to 2047). 
Engine uses automatic values (512-1536 MB) w/o maxMem parameter. 
```
https://community.bistudio.com/wiki/Arma_3_Startup_Parameters

**Comment by robalo** [comment]
Yeah, having experienced crashes myself until upping maxmem, I don't trust that wiki page. Also the official launcher says you can set up to 32 GB for it.

**Comment by havena** [comment]
dwarden confirme me, a month ago, the 2047 limit... i will try more.
i think 32 is for client.

**Comment by Dwarden** [comment]
2047 is still hardcoded maximum, 2.5 or 3GB is being tested internally only
(once there was uplift to 3GB but that's long gone cause it caused random crashes)

**Comment by havena** [comment]
here 2 last crash stats : http://i.imgur.com/eoXKN75.png

**Comment by havena** [comment]
@Killswitch00 dwarden say "from the full memory dump we know it crashed exactly on handling the _cfgEntry value but it makes no sense
value looks ok, also your playercount comment (only crash if 80+ makes the crash to be senseless even more)" (fnc_init_enum)

**Comment by havena** [comment]
i have fix some Extended_Init_Eventhandlers and try without cba_enable_auto_xeh

**Comment by ViperMaul** [comment]
@havena 
Status on crash?

**Comment by Killswitch00** [comment]
We cannot analyze and fix this crash, since it happens when the game runs the `configName` command with a valid input parameter.

**Comment by havena** [comment]
look like fixed since 1.50 / remove cba_enable_auto_xeh.pbo

**Comment by Killswitch00** [comment]
Crossing fingers that it stays fixed. :-)
